src/components/NanoContract/SelectAddressModal.component.js

```diff
@@ -69,14 +73,18 @@ export const SelectAddressModal = ({ address, show, onSelectAddress, onDismiss }
   };
 
   useEffect(() => {
-    const fetchData = async () => {
-      const iterator = await wallet.getAllAddresses();
-      const allAddresses = await consumeAsyncIterator(iterator);
-      setAddresses(allAddresses);
+    const fetchAllAddresses = async () => {
```

Comment:
Same here related to using sagas do deal with http requests

Reply:
Refactored.